flake.nix

```diff
@@ -38,447 +34,456 @@
-                before = lib.mkDefault mount.before;
-                wantedBy = mount.wantedBy or [ "multi-user.target" ];
-              })
-              options.mounts);
-          })
-
-          #################################################################### SSH (system level)
-          (mkIf true {
-            services.openssh = {
-              enable = true;
-              settings.PasswordAuthentication = false;
-              hostKeys = [{
-                path = options.ssh.privateKeyPath;
-                type = "ed25519";
-              }];
-            };
-          })
-
-          #################################################################### USER (core)
-          (mkIf true {
-            services.getty.autologinUser = "core";
-            users.users.core = {
-              isNormalUser = true;
-              group = "core";
-              extraGroups = [ "wheel" ];
-              openssh.authorizedKeys.keys = options.user.authorizedKeys;
-              shell = pkgs.fish;
-            };
-            users.groups.core = { };
-            environment.shells = [ pkgs.fish ];
-            programs.fish.enable = true;
-
-            home-manager.users.core = { pkgs, ... }: {
-
-              sops = {
-                defaultSopsFile = ./secrets/default.yaml;
-                secrets."wireguard/wg0" = {
-                  path = "%r/wireguard/wg0.conf";
-                };
-                age.sshKeyPaths = [ options.ssh.privateKeyPath ];
+              timezone = mkOption {
+                type = types.str;
+                default = "Europe/Helsinki";
+                description = "The time zone used when displaying times and dates";
               };
-
-              home.packages = with pkgs; [
-                file
-                tree
-                bind # nslookup
-              ];
-
-              programs = {
-                tmux.enable = true;
-                htop.enable = true;
-                vim.enable = true;
-                git.enable = true;
-                fish.enable = true;
-                fish.loginShellInit = "fish_add_path --move --prepend --path $HOME/.nix-profile/bin /run/wrappers/bin /etc/profiles/per-user/$USER/bin /run/current-system/sw/bin /nix/var/nix/profiles/default/bin";
-
-                home-manager.enable = true;
-              };
-              home.stateVersion = "23.05";
             };
-          })
-
-          #################################################################### WIREGUARD (no options)
-          (mkIf true {
-            systemd.services.wg0 = {
-              enable = true;
 
-              description = "wireguard interface for cross-node communication";
-              requires = [ "network-online.target" ];
-              after = [ "network-online.target" ];
+            mounts = mkOption {
+              type = types.attrsOf types.attrs;
+              default = { };
+              description = "Systemd mounts configuration";
```

Comment:
Also, I think that the module option mounts are still broken. 😮‍💨 

Nix repl (main):
```
[I] kari@torque ~/D/t/homestaking-infra (main) $ nix repl
Welcome to Nix 2.15.0. Type :? for help.

nix-repl> :lf .#
Added 22 variables.

nix-repl> outputs.nixosConfigurations.ponkila-ephemeral-beta.config.systemd.mounts
[ { ... } { ... } { ... } { ... } ]
```


Nix repl (this PR):
```
[I] kari@hetzner-ephemeral-alpha ~/homestaking-infra (juuso/config-extraction)> nix repl
Welcome to Nix 2.13.3. Type :? for help.

nix-repl> :lf .#
Added 22 variables.

nix-repl> outputs.nixosConfigurations.ponkila-ephemeral-beta.config.systemd.mounts
[ { ... } ]
```